docs/tracking.md

```diff
@@ -1 +1,1030 @@
+surface is reached, the active volume is switched, and the next set of layers
+is processed.
+
+Care has to be taken to correctly model the passive material, that is
+initially discarded with non-sensitive elements. For the material effects to
+be correctly taken into account during particle propagation, the material is
+projected onto dedicated material surfaces. These material surfaces are
+spread across the detector geometry. Each layer is created with two
+*approach surfaces* on either side. Their distance can be interpreted as
+the thickness of the layer in question. Examples of these approach surfaces
+can be found in {numref}`geometry_detail`, at the inner and outer radius.
+Approach surfaces, and the boundary surfaces between volumes mentioned before,
+are candidates to receive a projection of the surrounding material.
+Additional artificial material layers can also be inserted to receive
+projected material.
+
+The projection procedure works by extrapolating test particles using the
+fully detailed simulation geometry. During the extrapolation, the material
+properties of the geometry are sampled in small intervals. Subsequently, the
+same test particle is extrapolated through the tracking geometry. All
+material samples are then assigned and projected onto the closest material
+surface. Finally, the projection is averaged. The exact number and placement
+of the material surfaces has to be optimized to yield a sufficiently accurate
+representation of the inactive material in the detector.
+
+The numerical integration uses these projected material surfaces. Whenever
+such a surface is encountered in the propagation, the material properties are
+retrieved, and the corresponding modifications to the trajectory are
+executed. In case material is supposed to be integrated in a continuous way
+(as mentioned in [](#particle-propagation)), volumes can also store an
+effective volumetric material composition, which is queried by the numerical
+integration when needed. As the actual physical location of the detection
+hardware can vary over time, possible misalignment of the sensors needs to be
+handled correctly (see [](#alignment-of-particle-sensors)).
+
+## Clusterization
+
+The actual track reconstruction procedure itself starts with the conversion of
+raw inputs that have been read out from the detector. In case of silicon
+detectors, the readout can either be performed in a binary way, only recording
+which segments fired, or the amount of charges measured in the segment can be
+recorded, e.g. via *time-over-threshold* readout. In all cases, the readout is
+attached to an identifier uniquely locating the segment on the corresponding
+sensor.
+
+As a next step, these raw readouts need to be *clustered*, in order to
+extract an estimate of where particles intersect with the sensor. The general
+strategy of clustering algorithms follows the Connected Component Analysis (CCA)
```

Comment:
Also here we could link the clustering / clustering documentation.

Reply:
I don't think we were concretely planning this, but I can create a stub. @gagnonlg would you be able to fill that out in a follow-up PR?